crates/services/gas_price_service/src/lib.rs

```diff
@@ -152,21 +186,43 @@ where
                 tracing::debug!("Stopping gas price service");
                 should_continue = false;
             }
-            new_algo = self.update_algorithm.next() => {
-                let new_algo = new_algo?;
+            da_block_costs = self.da_block_costs_provider.recv() => {
+                self.da_block_costs = match da_block_costs {
+                    Ok(da_block_costs) => Some(da_block_costs),
+                    Err(err) => {
+                        tracing::error!("Failed to get da block costs: {:?}", err);
+                        None
+                    }
+                };
+                should_continue = true;
+            }
+            l2_block = self.l2_block_source.get_l2_block() => {
+                let l2_block = l2_block?;
+                let da_block_costs = self.da_block_costs.clone();
```

Comment:
Maybe you want to do `take` instead of `clone`=)

Reply:
addressed in 2f33f7b0a